core/src/main/scala/org/broadinstitute/dsde/rawls/entities/base/ExpressionParser.scala

```diff
@@ -0,0 +1,231 @@
+package org.broadinstitute.dsde.rawls.entities.base
+
+import org.broadinstitute.dsde.rawls.model.{Attributable, AttributeName, AttributeString, ParsedMCExpressions}
+
+import scala.util.parsing.combinator.JavaTokenParsers
+import scala.language.higherKinds
+import scala.util.Try
+import scala.language.postfixOps
+import spray.json._
+import cats.syntax.functor._
+import cats.instances.try_._
+
+/**
+  *
+  * @tparam F effect type
+  * @tparam ExpressionContext a context object that a particular implementation can use for context things
+  * @tparam PipeType a query that results in the root entity's name, entity records and a sort ordering
+  */
+trait ExpressionParser[F[_], ExpressionContext, PipeType] extends JavaTokenParsers {
```

Comment:
This trait is almost entirely the now removed `SlickExpressionParser`. The generic types are new. The main purpose of this change is to make this code useful for data repo expression parsing which runs in a `Future` not a `DBIO`.